### PR TITLE
Add SetFieldType Type

### LINKED ISSUE #718: Proposal: `SetFieldType`
A simple shallow field type replace utility type

definition:
```ts
import type {Except} from './except';
import type {Simplify} from './simplify';

export type SetFieldType<BaseType, Keys extends keyof BaseType, NewType> =
  Simplify<
  Except<BaseType, Keys> &
  Record<Keys, NewType>
  >;
```

use case:
```ts
type MyModel = { id: number; createdAt: Date; updatedAt: Date; enabled: boolean }
type MyModelApi = SetFieldType<MyModel, 'createdAt' | 'updatedAt', string>
// MyModelApi == { id: number; createdAt: string; updatedAt: string; enabled: boolean }

const response = fetch<MyModelApi>("url")
```

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/sindresorhus) so you can upvote and help fund this issue.
- The funding will be given to active contributors.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/sindresorhus/type-fest/issues/718">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/sindresorhus/type-fest/issues/718/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by lucasteles**
Fixes #718

### CODE REVIEW

File: source/set-field-type.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import type {Except} from './except';
+import type {Simplify} from './simplify';
+
+/**
+Create a type that changes the given keys types.
```
Comment (sindresorhus) [inline source/set-field-type.d.ts:5]:
```suggestion
Create a type that changes the type of the given keys.
```

readme too

File: source/set-field-type.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Use-case: You want to change the type of one or more fields of a model.
```
Comment (sindresorhus) [inline source/set-field-type.d.ts:7]:
```suggestion
Use-cases:
- Creating variations of a base model.
- Fixing incorrect external types.
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by Emiyaaaaa** [comment]
https://github.com/sindresorhus/type-fest/issues/718#issuecomment-1773811329
@sindresorhus @lucasteles

File: source/set-field-type.d.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import type {Except} from './except';
+import type {Simplify} from './simplify';
+
+/**
+Create a type that changes the type of the given keys.
```
Comment (sindresorhus) [inline source/set-field-type.d.ts:5]: This should be the text in the readme.

File: source/set-field-type.d.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Use-cases:
+- Creating variations of a base model.
+- Fixing incorrect external types.
+
```
Comment (sindresorhus) [inline source/set-field-type.d.ts:10]: Add a tip about using the `Merge` type if you need to change multiple properties to different types.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Tests are failing.

File: source/set-field-type.d.ts (hunk 3 above)
Comment (lucasteles) [inline source/set-field-type.d.ts:5]:
```suggestion
Create a type that changes the given keys types.
```

**Comment by sindresorhus** [comment]
Sorry if it was not clear with https://github.com/sindresorhus/type-fest/pull/721#discussion_r1371611693. It should be the inverse. Apply that text to the readme too.

**Pull request merged**